sdk/handler.go

```diff
@@ -38,47 +38,45 @@ func (h Handler) Serve(l net.Listener) error {
 }
 
 // ServeTCP makes the handler to listen for request in a given TCP address.
-// It also writes the spec file on the right directory for docker to read.
+// It also writes the spec file in the right directory for docker to read.
 func (h Handler) ServeTCP(pluginName, addr string, tlsConfig *tls.Config) error {
-	return h.listenAndServe("tcp", addr, pluginName, tlsConfig)
+	return h.listenAndServe(newTCPListener(addr, pluginName, tlsConfig))
 }
 
 // ServeUnix makes the handler to listen for requests in a unix socket.
-// It also creates the socket file on the right directory for docker to read.
+// It also creates the socket file in the right directory for docker to read.
 func (h Handler) ServeUnix(systemGroup, addr string) error {
-	return h.listenAndServe("unix", addr, systemGroup, nil)
+	return h.listenAndServe(newUnixListener(addr, systemGroup))
+}
+
+// ServeWindows makes the handler to listen for request in a windows named pipe.
+// It also creates the spec file in the right directory for docker to read.
+func (h Handler) ServeWindows(addr, pluginName string, pipeConfig *WindowsPipeConfig) error {
+	return h.listenAndServe(newWindowsListener(addr, pluginName, pipeConfig))
 }
 
 // HandleFunc registers a function to handle a request path with.
 func (h Handler) HandleFunc(path string, fn func(w http.ResponseWriter, r *http.Request)) {
 	h.mux.HandleFunc(path, fn)
 }
 
-func (h Handler) listenAndServe(proto, addr, group string, tlsConfig *tls.Config) error {
-	var (
-		err  error
-		spec string
-		l    net.Listener
-	)
+type newListenerFunc func() (net.Listener, string, string, error)
+
+func (h Handler) listenAndServe(fn newListenerFunc) error {
+
+	listener, addr, spec, err := fn()
```

Comment:
I guess this was changed as Windows doesn't need a `*tls.Config` but instead uses a *WindowsPipeConfig. I'm not sure if I like the way this is implemented - but let's see what the other maintainers think.

I think if I'd written this, I'd have changed the signature of `listenAndServe` and replaced `tlsConfig *tls.Config` with `config interface{}`. Then added a type assertion in the `switch` block to make sure we've got the tight type of config for the requested protocol.